pkg/secretless/plugin/sharedobj/internal_plugins.go

```diff
@@ -24,7 +24,7 @@ func GetInternalPluginsFunc() (plugin.AvailablePlugins, error) {
 			"aws":          aws.GetHTTPPlugin(),
 			"basic_auth":   basicauth.GetHTTPPlugin(),
 			"conjur":       conjur.GetHTTPPlugin(),
-			"generic_http": generic.GetHTTPPlugin(),
+			"generic_http": generichttp.GetHTTPPlugin(),
```

Comment:
Nit: Should we change the other naming conventions to include the http suffix as well? Or change this back to just generic? It seems weird to have the 'http' suffix for just this one. I guess it's possible we add generic plugins elsewhere, but that being the case we should probably change the package names for anything in the http package.

Reply:
It's a good sign that your comments are all about decisions I was on the fence about.

So here's the argument for keeping it.  The correctness of a package name can only be determined by looking at how it's used by other packages.  In this, from above:
```go
newConnector, err := generichttp.NewNewConnector(
```
Let's make two comparisons.  First, with how I had before:
```go
newConnector, err := generic.NewNewConnector(
```
Hmmm... in this case, what is "generic".  I think a reader of the code would need to jump to definition or look at the import path to figure it out.

Now let's see how a reference to basic auth looks:
```go
info := basicauth.PluginInfo()
```
The ambiguity is not the same as in the generic case.  A typical dev may be expected to understand that `basicauth` is a term relating to HTTP Basic Authentication.  We may not expect an analogous understanding when reading `generic`.  Hence the additional `http` is warranted in one case and not the other.  

A counter-counterpoint might be: Anything ever using the generic http connector will already be an an HTTP context....

Wdyt?

Reply:
I'm probably with @BradleyBoutcher that it doesn't seem to fit in nicely with the rest. I think the usage of that package can be fine with renaming in case of ambiguity which almost seems expected in a lot of other codebases that have this nested layout (eg. `import ( generichttp ".../http/generic" )`):
- https://github.com/kubernetes/sample-controller/blob/master/main.go#L30-L31
- https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/cmd.go

Reply:
alright.  `generic` it is!